### PR TITLE
📝 Add fine-grained tokens

### PR DESCRIPTION
**Pull request by coyotte508**
cc @philschmid (recent convo with cloudflare) @Pierrci too

### CODE REVIEW

**Review by julien-c — APPROVED** [review]

File: docs/hub/security-tokens.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ User Access Tokens are the preferred way to authenticate an application or noteb
 
 Access tokens allow applications and notebooks to perform specific actions specified by the scope of the roles shown in the following:
 
+- `fine-grained`: tokens with this role can be used to provide fine-grained access to specific resources, such as a specific model, or models in a specific organization. This type of tokens is useful in production environments, as you can use your own token without sharing access to all your resources.
```
Comment (osanseviero) [inline docs/hub/security-tokens.md:14]:
```suggestion
- `fine-grained`: tokens with this role can be used to provide fine-grained access to specific resources, such as a specific model or models in a specific organization. This type of token is useful in production environments, as you can use your own token without sharing access to all your resources.
```

File: docs/hub/security-tokens.md [hunk 2]
```diff
@@ -72,4 +74,6 @@ We recommend you create one access token per app or usage. For instance, you cou
  
  This way, you can invalidate one token without impacting your other usages.
 
-We also recommend only giving the appropriate role to each token you create. If you only need read access (e.g., loading a dataset with the `datasets` library or retrieving the weights of a model), only give your access token the `read` role.
+We also recommend only using fine-grained tokens for production usage. The impact if leaked will be reduced, and they can be shared among your organization without impacting your account.
```
Comment (osanseviero) [inline docs/hub/security-tokens.md:77]:
```suggestion
We also recommend only using fine-grained tokens for production usage. The impact, if leaked, will be reduced, and they can be shared among your organization without impacting your account.
```

**Review by osanseviero — APPROVED** [review]
Nice :fire:

**Review by pcuenca — APPROVED** [review]
Very cool!

File: docs/hub/security-tokens.md [hunk 3]
```diff
@@ -72,4 +74,6 @@ We recommend you create one access token per app or usage. For instance, you cou
  
  This way, you can invalidate one token without impacting your other usages.
 
-We also recommend only giving the appropriate role to each token you create. If you only need read access (e.g., loading a dataset with the `datasets` library or retrieving the weights of a model), only give your access token the `read` role.
+We also recommend only using fine-grained tokens for production usage. The impact, if leaked, will be reduced, and they can be shared among your organization without impacting your account.
+
+For example, if your production application needs read access to a gated model, a member of your organization can request access to the model and then create a fine-grained token with read access to that model. This token can then be used in your production application without giving it access to all your private models.
```
Comment (Pierrci) [inline docs/hub/security-tokens.md:79]: 🔥

**Pull request merged**